clients/cobol-dialect-api/src/v2api.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2025 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+
+import * as vscode from "vscode";
+import { type CopyStatementParser, MainExtensionId } from "./common";
+
+let v2Api: V2Api | undefined = undefined;
+
+/**
+ * Retrieve Version 1 of the API
+ * @param id id of the extension requesting v1 API
+ */
+export const getV2Api = async (extensionId: string): Promise<V2Api | Error> => {
+  if (!v2Api) {
+    const main = vscode.extensions.getExtension<unknown>(MainExtensionId);
+    if (main === undefined) {
+      return new Error(
+        `Cannot find COBOL Language Support extension ${MainExtensionId}`,
+      );
+    }
+    const api = await main.activate();
+    if (
+      typeof api !== "object" ||
+      api === null ||
+      !("version" in api) ||
+      typeof api.version !== "string"
+    ) {
+      return new Error(
+        `The version ${main.packageJSON.version} of COBOL Language Support extension ${main.id} you have installed does not provide API.`,
+      );
+    }
+    if (!("v2" in api)) {
+      return new Error(
+        `The version ${main.packageJSON.version} of COBOL Language Support extension ${main.id} you have installed does not provide v2 API: `,
+      );
+    }
+    const v2 = api.v2;
+    if (!isV2Api(v2)) {
+      return new Error(
+        `The v2 API of the version ${main.packageJSON.version} of COBOL Language Support extension ${main.id} you have installed does not match the expected signature.`,
+      );
+    }
+    v2Api = {
+      registerDialect: async (
+        dialect: V2DialectDetail,
+        handler: V2StartProcessingHandler,
+      ): Promise<vscode.Disposable | Error> => {
+        try {
+          return await v2.registerDialect(
+            extensionId,
+            {
+              name: dialect.name,
+              description: dialect.description,
+              snippets: dialect.snippets.toString(),
```

Comment:
We already have Uri here, just forward it.
```suggestion
              snippets: dialect.snippets,
```